### PR TITLE
Add org.jdownloader.JDownloader

### PR DESCRIPTION
**Pull request by eszlari**

### CODE REVIEW

**Comment by julianrichen** [comment]
I don't believe flatpak supports SVN but github manages a git mirror so this could be built from source:
https://github.com/mirror/jdownloader

Given the lack of a README/INSTALL I'm not sure how complicated it is to build/install.

**Comment by TingPing** [comment]
@julianrichen flatpak-builder 0.10.10 support svn

**Comment by eszlari** [comment]
According to to https://en.wikipedia.org/wiki/JDownloader not all parts are open source.

**Comment by barthalion** [comment]
It should be using extra-data, probably after we get external data checker integrated into CI. Downloading entire app (even if it's just one file) on every startup sounds wrong.

**Comment by nedrichards** [comment]
Er, yes - this is just wrong. Is there a EULA or similar licence for the downloaded binaries? That will tell you what the best option for dealing with them is.

Otherwise, obviously building from source is much better.

**Comment by eszlari** [comment]
> It should be using extra-data

As I said, it is too frequently updated to make this practical.

> Downloading entire app (even if it's just one file) on every startup sounds wrong.

It is only downloaded once, at the first startup.

**Comment by eszlari** [comment]
Okay, I looked at it again, maybe compiling is feasible, the closed parts seems to be in the plugins, that are automatically downloaded. The compiled JDownloader.jar than would have to be copied from /app to ~/.var on the first start, to make auto update work.

Now I just need to figure out how to compile it. Has anyone an idea, what build system this is:
https://github.com/mirror/jdownloader/blob/master/build/newBuild/build.xml

**Comment by julianrichen** [comment]
@eszlari If their docs are up-to-date then they recommend eclipse so maybe eclipse build file?
http://jdownloader.org/knowledge/wiki/development/get-started

But it looks like Apache ANT build system, example:
http://ant.apache.org/manual/tutorial-HelloWorldWithAnt.html

**Comment by eszlari** [comment]
I guess this has to be merged before:
https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk9/pull/7

**Comment by eszlari** [comment]
So, compiling will not work, because there seems to be some closed source dependency that is nowhere to be found on the Internet:
```
<property name="dep.jdc" value="../JDClosed" />
```
Either way, I read on their forums, that auto update will not work with self-compiled builds, so this would be useless anyway (because of the frequency, updating is important).

But I found URLs for binaries that seem to be rather stable (not changed since 2016), so I switch the manifest to using extra-data.

**Comment by nedrichards** [comment]
bot, build org.jdownloader.JDownloader

**Comment by eszlari** [comment]
ping?

**Comment by barthalion** [comment]
I'm still conflicted about the way it works, but given we already have applications that do the same, I'll merge it later today.

File: org.jdownloader.JDownloader.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  <id>org.jdownloader.JDownloader.desktop</id>
+  <name>JDownloader</name>
+  <metadata_license>CC0-1.0</metadata_license>
+  <summary>Download management tool</summary>
+  <description>
+    <p>
+      JDownloader is a free, open-source download management tool
+      with a huge community of developers that makes downloading
+      as easy and fast as it should be. Users can start, stop or
+      pause downloads, set bandwith limitations, auto-extract
```
Comment (AsciiWolf) [inline org.jdownloader.JDownloader.appdata.xml]: Typo: bandwith -> bandwidth

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.jdownloader.JDownloader

**Pull request closed without merging**